cmd/buildah/push.go

```diff
@@ -98,16 +103,47 @@ func pushCmd(c *cobra.Command, args []string, iopts pushResults) error {
 		return errors.New("Only two arguments are necessary to push: source and destination")
 	}
 
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	// Map of image tags found for the specified image.
+	// tags stored in the key
+	var imageTags []string
+	if iopts.allTags {
+		if err := getAllTags(src, &imageTags, store); err != nil {
+			return err
+		}
+		logrus.Debugf("For the image: [%s] found tags: %v", src, imageTags)
+	}
+
+	var errs *multierror.Error
+	imageName := src
+	destTarget := destSpec
 	compress := imagebuildah.Gzip
 	if iopts.disableCompression {
 		compress = imagebuildah.Uncompressed
 	}
 
-	store, err := getStore(c)
-	if err != nil {
-		return err
+	if len(imageTags) > 1 {
+		for _, tagKey := range imageTags {
+			src = imageName + ":" + tagKey
+			destSpec = destTarget + ":" + tagKey
```

Comment:
(\*shrug\* This looks fragile, but if `dest` can be an arbitrary destination, it’s a practical compromise I guess.)